alpha/apps/kaltura/lib/batch2/kFlowHelper.php

```diff
@@ -3207,41 +3207,45 @@ public static function handleBulkDownloadRetried(BatchJob $dbBatchJob, kBulkDown
 		$dbBatchJob->save();
 	}
 
-    public static function handleLiveToVodFinished(BatchJob $dbBatchJob, kLiveToVodJobData $data)
-    {
-        KalturaLog::info('Handling live to vod finished');
-        $liveEntryId = $data->getLiveEntryId();
-        $vodEntryId = $data->getVodEntryId();
-        $liveEntry = entryPeer::retrieveByPK($liveEntryId);
-        /** @var LiveStreamEntry $liveEntry */
-        $recordStatus = $liveEntry->getRecordStatus();
-        $shouldAutoArchive = $liveEntry->getRecordingOptions() ?
-			$liveEntry->getRecordingOptions()->getShouldAutoArchive() : false;
-        if ($recordStatus == RecordStatus::PER_SESSION && $shouldAutoArchive == true)
-        {
-            $liveEntryArchiveJobData = new kLiveEntryArchiveJobData();
-            $liveEntryArchiveJobData->setLiveEntryId($liveEntryId);
-            $liveEntryArchiveJobData->setVodEntryId($vodEntryId);
-
-            $liveEntryArchiveJob = new BatchJob();
-            $liveEntryArchiveJob->setEntryId($liveEntryId);
-            $liveEntryArchiveJob->setPartnerId($liveEntry->getPartnerId());
-
-            KalturaLog::info('Adding a job for auto archive');
-            kJobsManager::addJob($liveEntryArchiveJob, $liveEntryArchiveJobData, BatchJobType::LIVE_ENTRY_ARCHIVE);
-        }
-        return $dbBatchJob;
-
-    }
-
-    public static function handleKuserKgroupStatusUpdate($kuserkgroup)
-    {
-        if ($kuserkgroup->getStatus() == KuserKgroupStatus::DELETED)
-        {
-            $kgroup = kuserPeer::retrieveByPK($kuserkgroup->getKgroupId());
-            $numberOfUsersPerGroup = $kgroup->getMembersCount();
-            $kgroup->setMembersCount(max(0, $numberOfUsersPerGroup - 1));
-            $kgroup->save();
-        }
-    }
+	public static function handleLiveToVodFinished(BatchJob $dbBatchJob, kLiveToVodJobData $data)
+	{
+		KalturaLog::info('Handling live to vod finished');
+		$liveEntryId = $data->getLiveEntryId();
+		$vodEntryId = $data->getVodEntryId();
+		$liveEntry = entryPeer::retrieveByPK($liveEntryId);
+		/** @var LiveStreamEntry $liveEntry */
+		if (!$liveEntry){
+			throw new APIException(APIErrors::ENTRY_ID_NOT_FOUND, $liveEntryId);
+		}
+		else {
```

Comment:
no need for this else, throw is like return and will stop the running of the metho